pkg/handlers/generic/lifecycle/config/cm.go

```diff
@@ -92,14 +112,59 @@ func (h *HelmChartGetter) For(
 	}
 	d, ok := cm.Data[string(name)]
 	if !ok {
-		return nil, fmt.Errorf(
-			"did not find key %q in configmap %s/%s",
+		return nil, HelmChartInfoNotFoundError{
 			name,
 			h.cmNamespace,
 			h.cmName,
-		)
+		}
 	}
 	var settings HelmChart
 	err = yaml.Unmarshal([]byte(d), &settings)
 	return &settings, err
 }
+
+// For returns the HelmChart info for the given component from the configmap referenced in the cluster variables.
+// It first checks the configmap referenced in the cluster variables.
+// If not found, it returns the HelmChart info from the default configmap.
+func (h *HelmChartGetter) For(
+	ctx context.Context,
+	log logr.Logger,
+	cluster *clusterv1.Cluster,
+	name Component,
+) (*HelmChart, error) {
+	varMap := variables.ClusterVariablesToVariablesMap(cluster.Spec.Topology.Variables)
+
+	helmChartConfig, err := variables.Get[v1alpha1.HelmChartConfig](
+		varMap,
+		v1alpha1.ClusterConfigVariableName,
+		"addons", "helmChartConfig")
+	if err != nil {
+		if variables.IsNotFoundError(err) {
+			// HelmChartConfig is not defined in the cluster. Get the HelmChart from the default configmap.
+			return h.getInfoFor(ctx, log, name)
+		}
+		return nil, err
+	}
+	// helmChartConfig.ConfigMapRef or helmChartConfig.ConfigMapRef.Name will not be nil
+	// because it will be validated by the schema validation.
+	cmNameFromVariables := helmChartConfig.ConfigMapRef.Name
+	clusterNamespace := cluster.Namespace
+	hcGetter := NewHelmChartGetterFromConfigMap(cmNameFromVariables, clusterNamespace, h.cl)
+	overrideHelmChart, err := hcGetter.getInfoFor(ctx, log, name)
```

Comment:
helmChartFromCustomConfigMap would be a better name IMO. I had to think about what "override" meant in this context